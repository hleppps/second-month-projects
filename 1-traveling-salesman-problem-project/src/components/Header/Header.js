import styles from "./Header.module.css";

const Header = (props) => {
  return (
    <header className={styles.header}>
      <h1>TRAVELLING SALESMAN PROBLEM</h1>
      <button data-testid="button" className={styles.logo} onClick={props.changeThemeHandler}></button>
    </header>
  );
};

export default Header;
