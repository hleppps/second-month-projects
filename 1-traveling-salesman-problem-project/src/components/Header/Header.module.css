@import url(../../_variables.css);

.header {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 5rem;
  background-color: var(--selected-primary-color);
  color: var(--selected-secondary-color);
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0 5%;
  box-shadow: 0 2px 8px rgba(0, 0, 0, 0.25);
  z-index: 10;
}

.header button {
  padding: 1.5rem;
  border: none;
  border-radius: 50%;
  cursor: pointer;
  background-color: var(--selected-secondary-color);
}

.header h1 {
  user-select: none;
}