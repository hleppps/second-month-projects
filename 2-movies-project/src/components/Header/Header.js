import { useContext } from "react";
import { Button, Container, Navbar } from "react-bootstrap";
import AuthContext from "../../store/auth-context";

const Header = ({ onShowModal }) => {
  const { isLoggedIn, logout, userName } = useContext(AuthContext);

  return (
    <Navbar bg="dark" variant="dark">
      <Container>
        <Navbar.Brand className="user-select-none">Movies Project</Navbar.Brand>
        {!isLoggedIn && (
          <Button
            data-testid="login-button"
            variant="outline-light"
            onClick={onShowModal}
          >
            Login
          </Button>
        )}
        {isLoggedIn && (
          <Button
            data-testid="logout-button"
            variant="outline-light"
            onClick={() => {
              logout(false);
            }}
          >
            <span>Hello, </span>
            <span className="text-decoration-underline">{userName}</span>
            <span>!</span>
          </Button>
        )}
      </Container>
    </Navbar>
  );
};

export default Header;
