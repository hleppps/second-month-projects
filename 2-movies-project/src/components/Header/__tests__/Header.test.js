import { render, cleanup, fireEvent } from "@testing-library/react";
import AuthContext from "../../../store/auth-context";
import Header from "../Header";

const renderWithContext = (component, providerProps) => {
  return {
    ...render(
      <AuthContext.Provider value={providerProps}>
        {component}
      </AuthContext.Provider>
    ),
  };
};

afterEach(cleanup);

describe("Header component", () => {
  it("shows the modal", () => {
    const onShowModal = jest.fn();
    const isLoggedIn = false;

    const { getByTestId } = renderWithContext(
      <Header onShowModal={onShowModal} />,
      { isLoggedIn }
    );

    fireEvent.click(getByTestId("login-button"));
    expect(onShowModal).toBeCalled();
  });

  it("calls the logout function", () => {
    const logout = jest.fn();
    const isLoggedIn = true;

    const { getByTestId } = renderWithContext(<Header />, {
      isLoggedIn,
      logout,
    });

    fireEvent.click(getByTestId("logout-button"));
    expect(logout).toBeCalled();
  });

  it("matches a snapshot", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment(<Header />)).toMatchSnapshot();
  });
});
