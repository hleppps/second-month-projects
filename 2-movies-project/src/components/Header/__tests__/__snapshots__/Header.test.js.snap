// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches a snapshot 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand navbar-dark bg-dark"
  >
    <div
      class="container"
    >
      <span
        class="user-select-none navbar-brand"
      >
        Movies Project
      </span>
      <button
        class="btn btn-outline-light"
        data-testid="login-button"
        type="button"
      >
        Login
      </button>
    </div>
  </nav>
</DocumentFragment>
`;
